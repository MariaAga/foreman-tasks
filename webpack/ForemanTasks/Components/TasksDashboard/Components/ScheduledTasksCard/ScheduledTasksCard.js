import React from 'react';
import PropTypes from 'prop-types';
import { Card, Icon } from 'patternfly-react';
import classNames from 'classnames';

import {
  SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS_ARRAY,
  SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS,
} from './ScheduledTasksCardConstants';
import './ScheduledTasksCard.scss';
import TasksDonutCard from '../TasksDonutCard/TasksDonutCard';

class ScheduledTasksCard extends React.Component {
  onClick = () => {
    this.props.updateQuery({
      state: 'scheduled',
    });
  };
  render() {
    const { className, scheduled, focusedOn } = this.props;

    return (
      <Card
        className={classNames(
          'tasks-donut-card',
          'scheduled-tasks-card',
          className
        )}
      >
        <Card.Title>Scheduled</Card.Title>
        <Card.Body>
          <div
            className={`scheduled-data ${
              focusedOn === SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS.NONE
                ? 'not-focused'
                : ''
            }`}
            onClick={this.onClick}
          >
            {scheduled}
            <Icon type="fa" name="clock-o" />
          </div>
        </Card.Body>
      </Card>
    );
  }
}

const filterTitle = obj => {
  const { title, ...newObj } = obj;
  return newObj;
};

ScheduledTasksCard.propTypes = filterTitle(TasksDonutCard.propTypes);
ScheduledTasksCard.defaultProps = filterTitle(TasksDonutCard.defaultProps);

ScheduledTasksCard.propTypes = {
  className: PropTypes.string,
  focusedOn: PropTypes.oneOf(SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS_ARRAY),
  scheduled: PropTypes.number,
  updateQuery: PropTypes.func.isRequired,
};

ScheduledTasksCard.defaultProps = {
  title: '',
  className: '',
  focusedOn: SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS.NORMAL,
  scheduled: 0,
};

export default ScheduledTasksCard;
