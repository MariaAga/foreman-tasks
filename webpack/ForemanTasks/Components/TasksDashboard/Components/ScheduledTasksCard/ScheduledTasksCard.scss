.scheduled-tasks-card {
  text-align: center;
  .not-focused {
    opacity: 0.3;
  }
  .scheduled-data {
    cursor: pointer;
    font-size: 40px;
    font-weight: 300;
    * {
      margin: 10px;
    }
    &:hover {
      .donut-title-big-pf,
      .donut-title-small-pf {
        font-weight: bold;
      }
    }
  }
}
