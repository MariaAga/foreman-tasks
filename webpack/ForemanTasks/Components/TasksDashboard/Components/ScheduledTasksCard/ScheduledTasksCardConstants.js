export const SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS = {
  NORMAL: 'normal', // normal-mode
  NONE: 'none', // unfocus-mode: another card is selected
};

export const SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS_ARRAY = Object.values(
  SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS
);
