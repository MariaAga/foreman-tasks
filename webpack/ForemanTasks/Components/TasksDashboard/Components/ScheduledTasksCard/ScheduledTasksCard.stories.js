import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number, text, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { withCardsDecorator } from '../../../../../stories/decorators';

import { SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS } from './ScheduledTasksCardConstants';
import ScheduledTasksCard from './ScheduledTasksCard';

storiesOf('TasksDashboard', module)
  .addDecorator(withKnobs)
  .addDecorator(withCardsDecorator)
  .add('ScheduledTasksCard', () => (
    <div>
      <link
        href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css"
        rel="stylesheet"
        integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN"
        crossOrigin="anonymous"
      />
      <ScheduledTasksCard
        scheduled={number('errorLast', 1)}
        timePeriod={text('timePeriod', '24h')}
        focusedOn={select(
          'focusedOn',
          SCHEDULED_TASK_CARD_FOCUSED_ON_OPTIONS,
          ScheduledTasksCard.defaultProps.focusedOn
        )}
        updateQuery={action('Update Query')}
      />
    </div>
  ));
