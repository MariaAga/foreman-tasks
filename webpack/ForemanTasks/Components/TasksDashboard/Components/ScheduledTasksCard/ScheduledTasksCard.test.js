import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import ScheduledTasksCard from './ScheduledTasksCard';

const createRequiredProps = () => ({
  scheduled: 1,
});

const fixtures = {
  'render with minimal props': {},
  'render with props': { ...createRequiredProps() },
};

describe('ScheduledTasksCard', () =>
  testComponentSnapshotsWithFixtures(ScheduledTasksCard, fixtures));
