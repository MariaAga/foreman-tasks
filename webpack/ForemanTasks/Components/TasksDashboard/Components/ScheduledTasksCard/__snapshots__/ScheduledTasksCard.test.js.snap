// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduledTasksCard render with minimal props 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card scheduled-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  >
    Scheduled
  </CardTitle>
  <CardBody
    className=""
  >
    <div
      className="scheduled-data "
    >
      0
      <Icon
        name="clock-o"
        type="fa"
      />
    </div>
  </CardBody>
</Card>
`;

exports[`ScheduledTasksCard render with props 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card scheduled-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  >
    Scheduled
  </CardTitle>
  <CardBody
    className=""
  >
    <div
      className="scheduled-data "
    >
      0
      <Icon
        name="clock-o"
        type="fa"
      />
    </div>
  </CardBody>
</Card>
`;
